### DISCUSSION TITLE
Experimental support for 32-bit devices and Android 7

### DISCUSSION THREAD

**Discussion by markqvist** [markqvist/Reticulum#123] [Sideband]
There's now an experimental build of Sideband available that supports Android versions down to 7.0, and runs on 32-bit SoCs as well. Please note that I do not currently have the ability to test this, and it may or may not work. It is currently just being offered as an experiment, and because it may be useful in certain situations.

Notifications are currently disabled for Android versions earlier than 8.0, but all other functionality *should* work.

If you try it out, please let me know here how it works on your particular device(s).

[Download the APK](https://github.com/markqvist/Sideband/releases/download/0.2.4/sideband-20221020-experimental.apk)

**Comment by cinorid** [comment]
@markqvist 
I've got downloaded this file and tried installing it, but got parsing the package error. I think it's related to minimum android API level.

**Reply by markqvist** [reply]
Thanks for reporting. Maybe I will have to de separate APKs, I am not sure if the old devices are able to parse APKs with multiple platforms included. Do you know anything more about that?

**Reply by markqvist** [reply]
Ah, wait. My fault, made an error when I merged my two different branches to make a unified build script. Give me a sec...

**Reply by markqvist** [reply]
It's compiling now, but it's gonna take 10 minutes or so on my slow computer since it needs to recompile all the libs for the new API target ;)

**Reply by markqvist** [reply]
Ok, here is the updated APK: https://github.com/markqvist/Sideband/releases/download/0.2.4/sideband-20221020-experimental.apk

Could you try it out and let me know how it goes?

(it should have versioncode 20221020)

**Reply by cinorid** [reply]
It's installing now, and started normally

**Reply by samuk** [reply]
That also installs fine

![Screenshot_20221020-212653_Sideband](https://user-images.githubusercontent.com/400875/197051393-01c426c8-5b79-4891-8046-08d913a796df.png)

I get this error though. Is that expected? Do I need to do some configuration?

![Screenshot_20221020-212719_Sideband](https://user-images.githubusercontent.com/400875/197051592-f3a17738-a42c-4889-9fba-c0fe0ce0e47f.png)

**Reply by markqvist** [reply]
Nice, that is looking good so far then.

Yes, the sync error is OK, at least if Sideband has not yet heard any annouces from propagation nodes on the network. As soon as it hears one (or if you manually set a propagation node address in the preferences), it should be able to sync.

Thanks again for providing the feedback, it really helps, especially since I don't have any 32-bit devices to test it with ;)

**Comment by samuk** [comment]
Installs fine on SM-G960F with /eos on Android 10.

**Reply by markqvist** [reply]
Nice, thanks for the feedback!

**Comment by Swissbandit** [comment]
Hi mark,thanks for the effort to engage older devices.
With this i was able to install sideband to run on an older pad.
It is a samsung tablet galaxy A  SM-T580 running Android 8.1.0

**Reply by markqvist** [reply]
Thanks for providing the feedback! Really good to hear. It seems like it is workin out ok so far. Would you mind testing if notifications work under Android 8.1.0? (A peer has to be marked as trusted for it to display notifications, btw)

**Comment by Swissbandit** [comment]
Its working as it should,get a notification and sound when a new message arrives. But the new sb version is not woking per i2p anymore(also on the newer devices)

**Reply by markqvist** [reply]
Thanks for the feedback, and very much for the notice about i2p as well. I was not aware of that. I will investigate it as quickly as possibly.
